### ISSUE TITLE
Support @Sensitive annotation for attributes

### ISSUE THREAD

**Issue by qza** [javers/javers#500]
Some parameters like credit-card numbers, passwords etc are sensitive by nature. Changes to such parameters should be tracked, but value should not be stored inside the snapshot state.

At the moment, this can be achieved by defining @Sensitive annotation and using ObjectAccessHook together with custom SnapshotState converter that would clear the value of the field. However, this requires more traversals and could be done better inside core module (JsonConverterBuilder.SkipFieldExclusionStrategy)

Please guide if this can be achieved in some better way without modifying core.

**Comment by bartoszwalacik** [comment]
If you store such field outside JaversRepository, it wouldn't be visible in JQL. 
So for Javers it would be kind of @Transient but with a possibility to register you callback when change occurred?

**Comment by qza** [comment]
Not sure about the meaning of the callback. The main thing is that sensitive data should not be stored and changes MUST be tracked, in pure context of audit trail and SOX requirements.

**Comment by qza** [comment]
I need to get into Javers more deeply, so we can actually talk :)

**Comment by bartoszwalacik** [comment]
So how could you actually track these changes without persisting them?

**Comment by qza** [comment]
That is true, comparing to snapshot state in store it's not possible. Feel free to close this issue, as sensitive data should be encrypted already and stored like that.

**Comment by bartoszwalacik** [comment]
I think, that JaVers might implement encrypting/decrypting of certain attributes, annotated with  `@Sensitive` (or maybe `@Encrypted`), probably using some RSA keys.
Encrypted attributes would be stored safely in JaversRepository.

I'm not an expert in cryptography, so I'm open to your design suggestions and then a PR if you want it soon.

**Comment by ArielGrabijas** [comment]
Are there any updates on this issue? I'm confused, is it actually possible to achieve this in Javers?

**Comment by bartoszwalacik** [comment]
status update:
this issue is the feature request, if you need support for hiding/ciphering sensitive data in Javers, you can contribute a pull request with this feature to javers, then after review, it could be merged

**Comment by rolandoisidoro** [comment]
Any development in this issue?

I've coded a little POC comparing Hibernate Envers and JaVers. The later proved to be a much simpler solution and gathered our preference, until we need to try it with column encryption.

Envers works smoothly when implementing JPA’s `AttributeConverter` along with the `@Converter` annotation. JaVers does not.

Is there any known workaround to prevent JaVers to store sensitive data in clear text in its snapshots?

**Comment by bartoszwalacik** [comment]
I don't know any workaround, it's an open source project, @rolandoisidoro you are encouraged to contribute
